transformations/multilingual_lexicon_perturbation/transformation.py

```diff
@@ -0,0 +1,100 @@
+Serbian (sr), Swati (ss), Sundanese (su), Swedish (sv), Swahili (sw), Tamil (ta), Thai (th), Tagalog (tl), Tswana (tn), Turkish (tr),
+Ukrainian (uk), Urdu (ur), Uzbek (uz), Vietnamese (vi), Wolof (wo), Xhosa (xh), Yiddish (yi), Yoruba (yo), Chinese (zh), Zulu (zu)
+"""
+
+FOLDER_PATH = '/'.join(os.path.abspath(__file__).split('/')[:-1])
+
+def perturb_sentence(lexicon_df, text, prob_mix=0.5, src_lang="en", trg_lang="zh", seed=0):
+    random.seed(seed)
+    l_df = lexicon_df.set_index(src_lang)
+
+    words = word_tokenize(text)
+    mixed_text = ""
+    for word in words:
+        # Add space between word
+        if mixed_text != "":
+            mixed_text += " "
+            
+        if word.lower() not in l_df.index:
+            mixed_text += word
+        else:
+            rand_prob = random.random()
+            if rand_prob < prob_mix:
+                plain_word = word.translate(str.maketrans('', '', string.punctuation)).strip().lower()
+
+                if plain_word == "":
+                    continue
+
+                if not word[0].isupper(): # lower case
+                    perturbed_word = l_df.loc[plain_word, trg_lang]
+                else:
+                    perturbed_word = l_df.loc[plain_word, trg_lang].capitalize()
+                    
+                mixed_text += perturbed_word
+            else:
+                mixed_text += word
+
+    # print(text, mixed_text)
+    return mixed_text
+
+class MultilingualLexiconPerturbation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION
+    ]
+    supported_languages = [
+        'af', 'am', 'ar', 'ast', 'az', 'ba', 'be', 'bg', 'bn', 'br', 'bs', 'ca', 'ceb', 'cs', 'cy',
+        'da', 'de', 'el', 'en', 'es', 'et', 'fa', 'ff', 'fi', 'fr', 'fy', 'ga', 'gd', 'gl', 'gu',
+        'ha', 'he', 'hi', 'hr', 'ht', 'hu', 'hy', 'id', 'ig', 'ilo', 'is', 'it', 'ja', 'jv', 'ka',
+        'kk', 'km', 'kn', 'ko', 'lb', 'lg', 'ln', 'lo', 'lt', 'lv', 'mg', 'mk', 'ml', 'mn', 'mr',
+        'ms', 'my', 'ne', 'nl', 'no', 'ns', 'oc', 'or', 'pa', 'pl', 'ps', 'pt', 'ro', 'ru', 'sd',
+        'si', 'sk', 'sl', 'so', 'sq', 'sr', 'ss', 'su', 'sv', 'sw', 'ta', 'th', 'tl', 'tn', 'tr', 
+        'uk', 'ur', 'uz', 'vi', 'wo', 'xh', 'yi', 'yo', 'zh', 'zu'
+    ]    
+    heavy = False
+
+    def __init__(self, seed=0, prob_mix=0.5, src_lang="en", trg_lang="zh"):
+        super().__init__(seed)
+        if src_lang not in self.supported_languages: 
+            raise ValueError(f'Invalid `src_lang` value "{src_lang}". Supported languages: {supported_languages}')
+            
+        if trg_lang not in self.supported_languages:
+            raise ValueError(f'Invalid `trg_lang` value "{trg_lang}". Supported languages: {supported_languages}')
+            
+        self.lexicon_df = pd.read_pickle(f'{FOLDER_PATH}/multilingual_lexicon_uncased.xz')
+        
+        self.prob_mix=prob_mix
+        self.src_lang=src_lang
```

Comment:
_src_lang_ and _tgt_lang_ was used for a machine translation setting. You could maybe define _mlt_src_lang_ and  _mlt_tgt_lang_ to keep the variables different?

Reply:
Sure, this one has been updated on the latest commit